### PR TITLE
ISPN-12873 Fix empty XML from the REST cache config endpoint

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-12873

### CODE REVIEW

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 1]
```diff
@@ -402,8 +403,11 @@ private RestResponse getDetailResponse(Cache<?, ?> cache) {
             break;
          default:
             ByteArrayOutputStream entity = new ByteArrayOutputStream();
-            ConfigurationWriter writer = ConfigurationWriter.to(entity).withType(accept).build();
-            registry.serialize(writer, null, Collections.singletonMap(cacheName, cacheConfiguration));
+            try (ConfigurationWriter writer = ConfigurationWriter.to(entity).withType(accept).build()) {
+               registry.serialize(writer, null, Collections.singletonMap(cacheName, cacheConfiguration));
+            } catch (Exception e) {
+               return CompletableFuture.completedFuture(responseBuilder.status(INTERNAL_SERVER_ERROR).build());
```
Comment (gustavonalle) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:409]:
```suggestion
               return CompletableFuture.completedFuture(responseBuilder.status(INTERNAL_SERVER_ERROR).entity(Util.getRootCause(e)).build());
```

**Review by gustavonalle — CHANGES_REQUESTED** [review]

**Comment by gustavonalle** [comment]
stopped by the checkstyle police :police_car:

**Comment by tristantarrant** [comment]
Fixed

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java (hunk 1 above)
Comment (tristantarrant) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java]: It's there now

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheV2ResourceTest.java [hunk 2]
```diff
@@ -60,7 +60,8 @@
 import org.testng.annotations.Test;
 
 @Test(groups = "functional", testName = "rest.CacheV2ResourceTest")
-public class CacheV2ResourceTest extends AbstractRestResourceTest {
+public class
```
Comment (pruivo) [inline server/rest/src/test/java/org/infinispan/rest/resources/CacheV2ResourceTest.java:63]: revert change?

Comment (tristantarrant) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheV2ResourceTest.java]: Fixed

**Comment by pruivo** [comment]
integrated! thanks @tristantarrant !

**Pull request merged**